src/controls/QskSliderSkinlet.cpp

```diff
@@ -136,15 +136,17 @@ QRectF QskSliderSkinlet::sampleRect(
 
     if( slider->orientation() == Qt::Horizontal )
     {
-        r.setTop( r.center().y() - size.height() / 2 );
+        const auto y = r.center().y() - size.height() / 2;
+        r.setTop( qMax( 0.0, y ) );
         const auto x = r.left() + slider->valueAsRatio( pos ) * r.width() - size.width() / 2;
-        r.setLeft( x );
+        r.setLeft( qMax( 0.0, x ) );
     }
     else
     {
-        r.setLeft( r.center().x() - size.width() / 2 );
+        const auto x = r.center().x() - size.width() / 2;
+        r.setLeft( qMax( 0.0, x ) );
         const auto y = r.bottom() - slider->valueAsRatio( pos ) * r.height() - size.height() / 2;
-        r.setTop( y );
+        r.setTop( qMax( 0.0, y ) );
```

Comment:
From the title of this pull request I guess it is about preventing a situation, where the application crashes. Wouldn't it be better to add defensive check where the crash happens instead of avoiding running into that code path ?

Reply:
Yeah apparently the crash happens when y is < 0, which is calculated in the line above.
This is triggered by the orientation being neither Vertical nor Horizontal, hence the size hint is 0. In theory it can still happen that y is < 0 with layouting etc; do you know how a defensive check could look like?

Reply:
There should be no crash only because assigning a value < 0 to a local variable. Guess the crash is because of processing the calculated rectangle somewhere else. Then the defensive check would be against the questionable rectangle, where the crash happens.

Actually it might be possible, that a scene graph ends up being outside of the bounding rectangle of the item - f.e when not having a proper layout code. Another hypothetical situation I can imagine is when working with transformations.

Reply:
I recently fixed a heap corruption from the box renderer: https://github.com/uwerat/qskinny/commit/b5c56f7f8043391d1236eefcf4510d54d9aee4a2

Maybe this was also the reason for your crash ?

Reply:
> I recently fixed a heap corruption from the box renderer: [b5c56f7](https://github.com/uwerat/qskinny/commit/b5c56f7f8043391d1236eefcf4510d54d9aee4a2)
> 
> Maybe this was also the reason for your crash ?

Yes, it turns out with this patch the crash does not appear anymore.